x-pack/plugins/security_solution/public/detections/components/callouts/need_admin_for_update_callout/index.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { memo } from 'react';
+import { CallOutMessage, CallOutPersistentSwitcher } from '../../../../common/components/callouts';
+import { useUserData } from '../../user_info';
+
+import * as i18n from './translations';
+
+const needAdminForUpdateRulesMessage: CallOutMessage = {
+  type: 'primary',
+  id: 'need-admin-for-update-rules',
+  title: i18n.NEED_ADMIN_CALLOUT_TITLE,
+  description: i18n.needAdminForUpdateCallOutBody(),
+};
+
+/**
+ * Callout component that lets the user know that an administrator is needed for performing
+ * and auto-update of signals or not. For this component to render the user must:
+ * - Have the permissions to be able to read "signalIndexMappingOutdated" and that condition is "true"
+ * - Have the permissions to be able to read "hasIndexManage" and that condition is "false"
+ *
+ * If the user has the permissions to see that signalIndexMappingOutdated is true and that
+ * hasIndexManage is also true, then the user should be performing the update on the page which is
+ * why we do not show it for that condition.
+ */
+const NeedAdminForUpdateCallOutComponent = (): JSX.Element => {
+  const [{ signalIndexMappingOutdated, hasIndexManage }] = useUserData();
+
+  const signalIndexMappingIsOutdated =
+    signalIndexMappingOutdated != null && signalIndexMappingOutdated;
+
+  const userHasIndexManage = hasIndexManage != null && hasIndexManage;
+
+  return (
+    <CallOutPersistentSwitcher
+      condition={signalIndexMappingIsOutdated && !userHasIndexManage}
+      message={needAdminForUpdateRulesMessage}
```

Comment:
Here `signalIndexMappingOutdated != null` and `hasIndexManage != null` means that they have been loaded and now are available for checking.

Considering that, I suspect that condition might be incorrect.

`signalIndexMappingIsOutdated && !userHasIndexManage` ===
`signalIndexMappingIsOutdated && !(hasIndexManage != null && hasIndexManage)` ===
`signalIndexMappingIsOutdated && hasIndexManage == null && !hasIndexManage)` where
`hasIndexManage == null && !hasIndexManage` gives true when we are in initial state,
which would lead to flickering.

I think this is the right one:

```ts
const signalIndexMappingIsOutdated =
    signalIndexMappingOutdated != null && signalIndexMappingOutdated;
const userDoesntHaveIndexManage = hasIndexManage != null && !hasIndexManage;

condition={signalIndexMappingIsOutdated && userDoesntHaveIndexManage}
```

Reply:
Great catch here. I agree. Just want to point out one additional use case which is:

The `null` can be set when it's in an initial condition as well as when the user does not have enough privileges to even determine if `signalIndexMappingOutdated` so `null` is a bit overloaded in that it can stay `null` when the user does not know if they have enough privileges to determine if the signals index mapping is outdated. It's kind of odd, right, because we might at some point want the user to know for sure rather than work around the limitation. But for now, that is the limitation.

For `hasIndexManage` that _could_ be the case maybe too where the user doesn't have enough privileges to determine if they do or do not have the ability to manage an index. I think that's more doubtful looking at the code on the backend.

Regardless, I think what you're saying here is better and more valid of having this:

```ts
const userDoesntHaveIndexManage = hasIndexManage != null && !hasIndexManage;
```

As we would only determine if the user doesn't have an index manage privileges and show the callout if the user has enough privileges to check for this as well as take into account the initial `null` case of initialization.

I will merge this fix in.